cluster-diagnosis/sysdumpcollector.py

```diff
@@ -195,11 +198,34 @@ def collect_cilium_configmap(self):
             subprocess.check_output(cmd, shell=True)
         except subprocess.CalledProcessError as exc:
             if exc.returncode != 0:
-                log.error("Error: {}. Unable to get cilium configmap yaml")
+                log.error("Error: {}. Unable to get cilium configmap yaml"
+                          .format(exc))
         else:
             log.info("collected cilium configmap yaml file: {}".format(
                 configmap_file_name))
 
+    def collect_cilium_secret(self):
+        secret_file_name = "cilium-etcd-secrets-{}.json".format(
+            utils.get_current_time())
+        cmd = "kubectl get secret cilium-etcd-secrets -n kube-system -o json"
+        try:
+            output = json.loads(subprocess.check_output(cmd, shell=True))
+            data = {}
+            for key, value in output.get('data').iteritems():
+                data[key] = "XXXXX"
```

Comment:
nitpick: Change XXXX to **** . That's a more common form to represent redacted data.